pkg/services/authz/zanzana/server/auth.go

```diff
@@ -0,0 +1,24 @@
+package server
+
+import (
+	"context"
+
+	"github.com/grafana/authlib/claims"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ReqWithNamespace interface {
+	GetNamespace() string
+}
+
+func authorize(ctx context.Context, r ReqWithNamespace) error {
+	c, ok := claims.From(ctx)
+	if !ok {
+		return status.Errorf(codes.Unauthenticated, "unauthenticated")
+	}
+	if !claims.NamespaceMatches(c.GetNamespace(), r.GetNamespace()) {
+		return status.Errorf(codes.PermissionDenied, "namespace does not match")
+	}
+	return nil
+}
```

Comment:
nit: to make this one simpler just take a `namespace string` instead of interface

Reply:
hm, what do you mean?

Reply:
Do you mean pass namespace as a parameter instead of whole request?